source/compiler/qsc_circuit/src/circuit.rs

```diff
@@ -214,6 +214,28 @@ impl Operation {
         }
     }
 
+    #[must_use]
+    pub fn targets_mut(&mut self) -> &mut Vec<Register> {
+        match self {
+            Operation::Measurement(m) => &mut m.qubits,
+            Operation::Unitary(u) => &mut u.targets,
+            Operation::Ket(k) => &mut k.targets,
+        }
+    }
+
+    #[must_use]
+    pub fn all_qubits(&self) -> Vec<Register> {
+        match self {
+            Operation::Measurement(m) => m.qubits.clone(),
+            Operation::Unitary(u) => {
+                let mut qubits = u.targets.clone();
+                qubits.extend_from_slice(&u.controls);
+                qubits
+            }
+            Operation::Ket(k) => k.targets.clone(),
+        }
+    }
+
```

Comment:
Please see above comment - if you use `OperationOrGroup` when manipulating the circuit, you shouldn't have to add these methods

Reply:
I did end up having to keep `targets_mut` as I needed a way to prune the targets of the operation representing a group.